app/models/version.js

```diff
@@ -49,7 +49,10 @@ export default class Version extends Model {
   }
 
   @cached get isFirst() {
-    let { versions } = this.crate;
+    let { versions, firstVersionId } = this.crate;
+    if (firstVersionId !== undefined) {
+      return this.id === firstVersionId;
+    }
```

Comment:
is the `else` branch possible? as in, is it possible for `firstVersionId` to be `null`? I assume it's possible if `this.versions` in the crate is empty, but I guess in that was we wouldn't be looking at a `Version` model for the crate?

Reply:
To be honest, my lack of familiarity with Ember.js means this approach might be more prone to falling back to the `else` branch if something unexpected happens. Ideally, with everything working as intended, we wouldn't need to use the `else` branch at all.

Reply:
I think I'd just make it `return this.id === this.crate?.firstVersionId`. Worst thing that could happen then is that `this.crate` or `this.crate.firstVersionId` are indeed undefined for some weird reason and in that case `isFirst` would be `false` 🤷‍♂️